x/go/signal/routine.go

```diff
@@ -55,6 +58,14 @@ const (
 	Panicked
 )
 
+//go:generate stringer -type=PanicPolicy
+const (
```

Comment:
If we're going to make this an enum, it might be worth to fix `cancelOnExit` and `cancelOnExitErr` to be an enum as well (albeit probably a private one so we can keep the same options pattern). This could get rid of the strange double boolean and extra `||` checks.

Reply:
Thoughts on renaming `cancelOnExitErr` to `cancelOnFail`?

Reply:
Sure. But you should make it an enum.